vectorizing_data/setup_milvus.py

```diff
@@ -0,0 +1,136 @@
+import json
+import torch
+import torchvision
+import torch.nn as nn
+from torchvision import transforms
+import argparse
+
+from tqdm import tqdm
+from PIL import Image
+from pathlib import Path
+from torch.utils.data import DataLoader, Dataset
+from pymilvus import connections, Collection, db, MilvusClient
+
+from transformers import CLIPProcessor, CLIPModel
+from pymilvus import FieldSchema, CollectionSchema, DataType, Collection, connections, db
+
+
+MILVUS_HOST = "localhost"
+MILVUS_PORT = "19530"
+
+DB_NAME = "data_embeddings"
+COLLECTION_NAME = "data_clip_embeddings"
+
+
+def read_json(file_path):
+    with open(file_path, 'r') as f:
+        return json.load(f)
+
+def create_db():
+    conn = connections.connect(host=MILVUS_HOST, port=MILVUS_PORT)
+    database = db.create_database(DB_NAME)
+
+
+def create_video_collection():
+    conn = connections.connect(host=MILVUS_HOST, port=MILVUS_PORT, db_name=DB_NAME)
+
+    fields = [
+        FieldSchema(name="id", dtype=DataType.INT64, is_primary=True, auto_id=True),
+        FieldSchema(name="embedding", dtype=DataType.FLOAT_VECTOR, dim=768),
+        FieldSchema(name="path", dtype=DataType.VARCHAR, max_length=255),
+    ]
+
+    schema = CollectionSchema(fields=fields, description="CLIP video embeddings collection")
+    collection = Collection(name=COLLECTION_NAME, schema=schema)
+    index_params = {
+        'metric_type': 'COSINE',
+        'index_type': 'IVF_FLAT',
+        'index_name': 'embeddings_index',
+        'params':{'nlist': 16384},
```

Comment:
how did you chose this?

Reply:
This is a standart value of 2**14, that is used in lots of examples. Based on my previous experiments and researches, it gives best results without using to much resources. So it was selected mostly heuristically based on my previous experience.